signer/signer.go

```diff
@@ -34,13 +38,16 @@ type UnsignedAttributes struct {
 
 // SignRequest is used to generate Signature.
 type SignRequest struct {
-	Payload             []byte
-	PayloadContentType  PayloadContentType
-	SignatureProvider   SignatureProvider
-	SigningTime         time.Time
-	Expiry              time.Time
-	ExtendedSignedAttrs []Attribute
-	SigningAgent        string
+	Payload                      []byte
+	PayloadContentType           PayloadContentType
+	SignatureProvider            SignatureProvider
+	SigningTime                  time.Time
+	Expiry                       time.Time
+	ExtendedSignedAttrs          []Attribute
+	SigningAgent                 string
+	SigningScheme                SigningScheme
+	VerificationPlugin           string
+	VerificationPluginMinVersion string  // TODO: Implement SimVer structure
```

Comment:
```suggestion
	VerificationPluginMinVersion string  // TODO: Implement SemVer structure
```

Reply:
We can accept a string here rather than a struct. The TODO part is validating that the SemVer string is valid

Reply:
As we need to validate semver which has its own semantics and compare with other semver, IMO it would be better to have well defined struct and fail early.

Reply:
Since golang doesnt provides require library ([semver in x pkg](https://pkg.go.dev/golang.org/x/mod@v0.5.1/semver)) doest abide by https://semver.org/. Used regular expression to validate semver.